packages/messenger-widget/src/hooks/haltDelivery/useHaltDelivery.ts

```diff
@@ -145,8 +155,13 @@ export const useHaltDelivery = () => {
 
             await submitEnvelopsToReceiversDs(dispatchableEnvelops);
 
+            // delete the halted messages with messageId
             dispatchableEnvelops.map((envelop) => {
-                clearHaltedMessages(envelop.haltedEnvelopId, envelop.aliasName);
+                // fetch the messageId from message signature
+                if (envelop.envelop.metadata?.signature) {
+                    const msgId = msgMap.get(envelop.envelop.message.signature);
```

Comment:
I think it would cause problems this way because the signatures from a dispatchable envelope is different from its original envelope. After all, the deliveryinformation is also part of the signature.

Reply:
We use message signature and not the envelop signature, so I don't think it should mess up because message signature should remain same.